### DISCUSSION TITLE
fromUnixTime vs parse vs toDate

### DISCUSSION THREAD

**Discussion by JeffBeltran** [js-fns/date-fns#2206] [Q&A] [answered]
This might be something really dumb that im missing but i would think these 3 methods would do the same thing

<img width="585" alt="image" src="https://user-images.githubusercontent.com/22965241/108286407-de36a900-713d-11eb-8382-58932145804f.png">

but it looks like `fromUnixTime` doesn't.

From the docs
- `fromUnixTime`: Create a date from a Unix timestamp.
- `toDate`: If the argument is a number, it is treated as a timestamp.
- `parse`: Milliseconds timestamp `T`

what am i missing?

**Comment by fturmel** [comment, marked as answer]
Hi @JeffBeltran,

Going through older discussions and I saw that no one answered you. It's probably a bit late, but maybe it'll help someone else. The difference here is simply that `fromUnixTime` expects a Unix timestamp in _seconds_ (not milliseconds).

You can see it in the example section of the docs, but it's not very obvious. This could be improved, I'll work on a PR. https://date-fns.org/docs/fromUnixTime

So going back to your example, these would be roughly equivalent except the `fromUnixTime` date will have its milliseconds component lost in the conversion:
```js
const ms = 1613611083646;

fromUnixTime(ms / 1000);
toDate(ms);
parse(ms, "T", new Date());
```

**Reply by JeffBeltran** [reply]
> something really dumb that im missing

yup, not sure why i didn't think to try that, but yeah time stuff is always weird. Thanks for the reply though
